modules/programs/silicon.nix

```diff
@@ -0,0 +1,131 @@
+{ pkgs, lib, config, ... }:
+
+with lib;
+
+let cfg = config.programs.silicon;
+in {
+  meta.maintainers = with hm.maintainers; [ afresquet ];
+
+  options.programs.silicon = {
+    enable =
+      mkEnableOption "silicon, create beautiful image of your source code";
+
+    package = mkPackageOption pkgs "silicon" { };
+
+    settings = mkOption {
+      type = types.str;
+      default = "";
+      example = literalExpression ''
+        --shadow-color '#555'
+        --background '#fff'
+        --shadow-blur-radius 30
+        --no-window-controls
+      '';
+      description = ''
+        Silicon configuration.
+      '';
+    };
```

Comment:
When I worked on my own module for Silicon, I went with a more true-to-Nix approach and I'd love to hear your thoughts. Settings would be set with Nix values (https://github.com/uncenter/flake/blob/9908d5d4f3e345459dfc638f6434d9d28ec5446b/modules/extra/home-manager/silicon.nix#L11-L23) and this attrset would then be converted into the configuration file with https://github.com/uncenter/flake/blob/9908d5d4f3e345459dfc638f6434d9d28ec5446b/modules/extra/home-manager/silicon.nix#L43-L57. You can see it in action at https://github.com/uncenter/flake/blob/9908d5d4f3e345459dfc638f6434d9d28ec5446b/home/programs/silicon.nix#L6-L10.